adapters/genericListChecker/builder_test.go

```diff
@@ -26,119 +26,56 @@ func TestBuilderInvariants(t *testing.T) {
 	testutil.TestBuilderInvariants(b, t)
 }
 
-func TestWhitelistWithOverrides(t *testing.T) {
+func runTests(t *testing.T, bc BuilderConfig, ac AdapterConfig, goodValues []string, badValues []string) {
```

Comment:
Why not just have one TestCheckList that loops over a struct that includes a name, BuilderConfig, AdapterConfig, goodValues, and badValues?

The struct could live outside the TestCheckList method.

I think that would be more idiomatic and a bit easier to read.